docs/developer/architecture/solutions.rst

```diff
@@ -0,0 +1,241 @@
+
+What is a Solution ?
+
+It's a packaged Kubernetes application, archived as an ISO disk image,
+containing:
+
+* A set of OCI images to inject in MetalK8s image registry
+* An Operator to deploy on the cluster
+* A UI for managing and monitoring the application
+
+.. _Helm: https://helm.sh/
+
+Requirements
+------------
+
+* Everything must work offline.
+* Solutions must still be available after a reboot.
+* Check archives integrity and validity.
+* Deploy Solutions (operator, UI, CRDs, StorageClass, Grafana dashboards).
+* Handle multiple Solutions with different versions.
+* Guidelines for Solutions developers.
+
+User Stories
+------------
+
+Solution import
+~~~~~~~~~~~~~~~
+
+As a cluster admin, I want to be able to import a Solution archive using a CLI
+tool, to make the Solution available for deployment.
+
+Environment creation/deletion/inspection
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As a cluster admin, I want to be able to create/delete/inspect an Environment
+using the CLI.
+
+Solution deployment
+~~~~~~~~~~~~~~~~~~~
+
+As a solution admin, I want to be able to deploy a Solution through its
+UI, without any specific knowledge about Kubernetes.
+
+Solution upgrade/downgrade
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As a solution admin, I want to be able to easily upgrade or downgrade a
+Solution using its UI without any specific knowledge about Kubernetes.
+
+Solution deletion
+~~~~~~~~~~~~~~~~~
+
+.. todo:: Open question:
+          do we want the user to be able to delete its solution ?
+          no ?
+          yes, but with some safety nest ?
+          Only with a tool from the CLI ?
+          If not, do we want to at least document the manual procedure ?
+
+Solution development
+~~~~~~~~~~~~~~~~~~~~
+
+As a developer, I want to have guidelines to follow to develop a Solution.
+
+Solution packaging
+~~~~~~~~~~~~~~~~~~
+
+As a developer, I want to have documentation to know how to package a Solution.
+
+Solution validation
+~~~~~~~~~~~~~~~~~~~
+
+As a developer, I want to be able to know that a packaged Solution
+follows the requirements and is valid using a CLI tool.
+
+Design Choices
+--------------
+
+Solution configuration
+~~~~~~~~~~~~~~~~~~~~~~
+
+We will use a ``SolutionsConfiguration`` object, stored in
+``/etc/metalk8s/solutions.yaml``, to declare which Solutions are available
+to the cluster, from the bootstrap node.
+
+Here is how it will look::
+
+    apiVersion: metalk8s.scality.com/v1alpha1
+    kind: SolutionsConfiguration
+    solutions:
+      - /solutions/storage_1.0.0.iso
+      - /solutions/storage_latest.iso
+      - /other_solutions/computing.iso
+
+There is no explicit information about what an archive contains.
+Instead, we want the archive itself to contain such information and to
+discover it at import time.
+
+This configuration file will be read by a Salt external pillar module,
+which will permit the consumption of it by Salt modules and states.
```

Comment:
Yes - let's add that it does the same as `BootstrapConfig` (file in `/etc/metalk8s`, mounted in the Salt master container...).

Let's also document the entire `metalk8s:solutions` pillar section in this design: what each key represents and how it's going to be computed.